addons/vagrant/playbooks/linux_servers/samba4ad.yml

```diff
@@ -0,0 +1,201 @@
+- hosts: service_samba4ad
+  name: Manage Samba service
+  become: True
+
+  collections:
+    - debops.debops
+    - debops.roles01
+    - debops.roles02
+    - debops.roles03
+
+  #roles:
+  #  - role: samba
+
+  tasks:
+  - name: Disable IPV6
+    ansible.posix.sysctl:
+      name: "{{ item }}"
+      value: '1'
+      sysctl_set: true
+    loop:
+    - "net.ipv6.conf.all.disable_ipv6"
+    - "net.ipv6.conf.default.disable_ipv6"
+    - "net.ipv6.conf.lo.disable_ipv6"
+
+  - name: Install Samba packages
+    ansible.builtin.package:
+      name: '{{ item }}'
+      state: 'present'
+    loop:
+    - gnupg
+    - lsb-release
+    - python3-pexpect
+
+  - name: Reboot
+    ansible.builtin.reboot:
+
+  - name: Wait for the reboot and reconnect 
+    wait_for_connection:
+
+  - name: Pause for 10 seconds after reboot
+    ansible.builtin.pause:
+      seconds: 10
+
+  - name: Download file with check (sha256)
+    ansible.builtin.get_url:
+      url: http://samba.tranquil.it/tissamba-pubkey.gpg
+      dest: /root/tissamba-pubkey.gpg
+      checksum: sha256:bd0f7140edd098031fcb36106b24a6837b067f1c847f72cf262fa012f14ce2dd
+
+  - name: Add samba gpg
+    ansible.builtin.apt_key:
+      file: /root/tissamba-pubkey.gpg
+      state: present
+    
+  - name: add apt source
+    ansible.builtin.apt_repository:
+      repo: "deb https://samba.tranquil.it/debian/samba-4.16/ bullseye main"
+      state: present
+
+  - name: update and upgrade
+    ansible.builtin.apt:
+      upgrade: yes
+      update_cache: yes
+
+  - name: Install Samba packages
+    ansible.builtin.package:
+      name: '{{ item }}'
+      state: 'present'
+    loop: '{{ samba__base_packages }}'
+
+  - name: Stop and Mask Services
+    ansible.builtin.systemd:
+      enabled: no
+      state: stopped
+      masked:  yes
+      name: "{{ item }}"
+    loop: "{{ samba__services_to_stop }}"
+
+  - name: Remove config files
+    file:
+      path: "{{ item }}"
+      state: absent
+    loop:
+      - "/etc/samba/smb.conf"
+      - "/etc/krb5.conf"
+
+  - name: change krb5.conf
+    ansible.builtin.blockinfile:
+      create: true
+      dest: /etc/krb5.conf
+      content: |
+         [libdefaults]
+            default_realm = "{{ dns_domain_upper }}"
+            dns_lookup_kdc = "{{ domain_setup__kdc }}"
+            dns_lookup_realm = "{{ domain_setup__realm }}"
+         
+         [realms]
+            "{{ dns_domain_upper }}" = {
+              "admin_server = AD.{{ dns_domain_upper }}"
+              "kdc = AD.{{ dns_domain_upper }}"
+         }
+         
+         [domain_realm]
+           "{{ dns_domain }} = {{ dns_domain_upper }}"
+           ".{{ dns_domain }} = {{ dns_domain_upper }}"
+
+  - name: Adding localhost in resolv.conf
+    ansible.builtin.lineinfile:
+      path: /etc/dhcp/dhclient.conf
+      line: prepend domain-name-servers 127.0.0.1, 8.8.8.8;
+      state: present
+      create: true
```

Comment:
The description doesn't seem to match what task does:
* description: adding localhost in resolv.conf
* task: add domain-name-servers in /etc/dhcp/dhclient.conf

Reply:
Well, in fact it is fixing the resolv.conf on reboot by surcharging the dhclient.conf.
It is like adding a default resolv.conf even if reboot.
It was the lightest way I found to override it, I am sure there are other way to do it.

Reply:
OK. I got the point. Could you add a comment to explain this ?